Templates/Integrations/AnalyticsAndSIEM/AnalyticsAndSIEM.py

```diff
@@ -0,0 +1,354 @@
+
+        Returns:
+            Response json
+        """
+        return self._http_request('GET', 'version')
+
+    def list_events_request(self, limit: Union[int, str] = None, since_time: Optional[str] = None) -> Dict:
+        """Gets all credentials from API.
+
+        Args:
+            limit: limit results
+            since_time: timestamp to start pull events from
+
+        Returns:
+            Response from API. from since_time if supplied else returns all events in given limit.
+        """
+        suffix = 'event'
+        params = assign_params(sinceTime=since_time, limit=limit)
+        return self._http_request('GET', suffix, params=params)
+
+    def event_request(self, event_id: AnyStr) -> Dict:
+        """Gets events by ID
+
+        Args:
+            event_id: event id to get
+
+        Returns:
+            event details
+        """
+        # The service endpoint to request from
+        suffix = 'event'
+        # Dictionary of params for the request
+        params = assign_params(eventId=event_id)
+        return self._http_request('GET', suffix, params=params)
+
+    def close_event_request(self, event_id: AnyStr) -> Dict:
+        """Closes event by ID
+
+        Args:
+            event_id: event to delete
+
+        Returns:
+            Response from API
+        """
+        # The service endpoint to request from
+        suffix = 'event'
+        # Dictionary of params for the request
+        params = assign_params(eventId=event_id)
+        # Send a request using our http_request wrapper
+        return self._http_request('DELETE', suffix, params=params)
+
+    def update_event_request(self, event_id: AnyStr, description: Optional[AnyStr] = None,
+                             assignee: Optional[List[str]] = None) -> Dict:
+        """Update given event
+
+        Args:
+            description: change description of event
+            assignee: User to assign event to
+            event_id: event ID
+
+        Returns:
+            Response from API
+        """
+        suffix = 'event'
+        params = assign_params(eventId=event_id, description=description, assignee=assignee)
+        return self._http_request('POST', suffix, params=params)
+
+    def create_event_request(self, description: str, assignee: List[str] = None) -> Dict:
+        """Creates given event.
+
+        Args:
+            description: change description of event
+            assignee: User to assign event to
+
+        Returns:
+            Response from API
+        """
+        suffix = 'event'
+        params = assign_params(description=description, assignee=assignee)
+        return self._http_request('POST', suffix, params=params)
+
+    def query_request(self, **kwargs):
+        suffix = 'query'
+        return self._http_request('GET', suffix, params=kwargs)
+
+
+''' HELPER FUNCTIONS '''
+
+
+def build_dict(event: Dict) -> Dict:
```

Comment:
name is too broad.
maybe `build_event_context`
also why do we need two functions? why not always return a list of event contexts (incase of single event, return list of one entry) 🤔